openeogeotrellis/layercatalog.py

```diff
@@ -972,6 +972,36 @@ def _merge_layers_with_common_name(metadata: CatalogDict):
     return metadata
 
 
+def query_jvm_opensearch_client(open_search_client, collection_id, _query_kwargs, processing_level=""):
+    jvm = get_jvm()
+    fromDate = jvm.java.time.ZonedDateTime.parse(str(_query_kwargs["start_date"]).replace(" ", "T") + "Z")
+    toDate = jvm.java.time.ZonedDateTime.parse(str(_query_kwargs["end_date"]).replace(" ", "T") + "Z")
+    dateRange = jvm.scala.Some(jvm.scala.Tuple2(fromDate, toDate))
+    extent = jvm.geotrellis.vector.Extent(float(_query_kwargs["ulx"]), float(_query_kwargs["bry"]),
+                                          float(_query_kwargs["brx"]), float(_query_kwargs["uly"]))
+    crs = jvm.geotrellis.proj4.CRS.fromEpsgCode(4326)
+    bbox = jvm.geotrellis.vector.ProjectedExtent(extent, crs)
+    attribute_values_dict = {}
+    if "cldPrcnt" in _query_kwargs:
+        attribute_values_dict["eo:cloud_cover"] = _query_kwargs["cldPrcnt"]
+    if "eo:cloud_cover" not in attribute_values_dict:
+        # Terrascope only has <95% clouded tiles, so use as default
+        # So we will not get missing warnings for totally clouded tiles.
+        attribute_values_dict["eo:cloud_cover"] = 95
```

Comment:
actually we cannot do this: if the user does not filter on cloud cover, the default is to include everything, and we should indeed switch to not use Terrascope

Reply:
removed it